packages/web-client/app/services/card-customization-options.ts

```diff
@@ -0,0 +1,91 @@
+import Service from '@ember/service';
+import { tracked } from '@glimmer/tracking';
+import config from '../config/environment';
+import {
+  ColorCustomizationOption,
+  PatternCustomizationOption,
+} from '../utils/web3-strategies/types';
+
+interface ColorCustomizationResponseObject {
+  attributes: {
+    'pattern-color': string;
+    'text-color': string;
+    background: string;
+    description?: string;
+  };
+  id: string;
+}
+
+interface PatternCustomizationResponseObject {
+  attributes: {
+    'pattern-url': string;
+    description?: string;
+  };
+  id: string;
+}
+
+let convertToColorCustomizationOption = (
+  o: ColorCustomizationResponseObject
+): ColorCustomizationOption => {
+  return {
+    patternColor: o.attributes['pattern-color'],
+    textColor: o.attributes['text-color'],
+    headerBackground: o.attributes.background,
+    description: o.attributes.description,
+    id: o.id,
+  };
+};
+
+let convertToPatternCustomizationOption = (
+  o: PatternCustomizationResponseObject
+): PatternCustomizationOption => {
+  return {
+    patternUrl: o.attributes['pattern-url'],
+    description: o.attributes.description,
+    id: o.id,
+  };
+};
```

Comment:
It would be common in Ember development, when faced with writing this code, to reach for a data library, usually ember-data or orbit.js. It avoids needing this kind of boilerplate deserialization (at the cost of a complex dependency). We can refactor to it later, just want to flag this.